startup/gui/editor.py

```diff
@@ -41,6 +41,12 @@ def __editorKeyPress( editor, event ) :
 	if event.key == "B" and event.modifiers == event.modifiers.Control :
 		return GafferUI.GraphBookmarksUI.popupFindBookmarkMenu( editor )
 
+	if event.key in map( str, range( 10 ) ) :
```

Comment:
When I did the "Ctrl+B" thing, I wasn't sure if I'd got the right API for GraphBookmarksUI. Specifically I wondered if it should just be more like a single call to opt in to everything for an application, like `DotUI.connect()`. For this new functionality, I'm feeling like maybe I got it wrong - it feels a bit weird to be calling these API methods that don't actually do anything in many cases (early returns based on editor type etc). I wonder if just `connect()` would be better, or `handleEditorKeyPress()` if we wanted finer-grained control.

I haven't had a consistent approach to this in the past, so we have various methods of doing this sort of thing :

- Some things just add their functionality unconditionally as soon as GafferUI is imported.
- Some things are opt-in via a config with a single call.
- Some things are opt-in via a config in a finer-grained way.

I'd value your thoughts on what's the right way in this case, and more generally...